File: mirai-core/src/main/java/net/mamoe/mirai/event/events/qq/FriendEvent.java
package net.mamoe.mirai.event.events.qq;

import net.mamoe.mirai.Bot;
import net.mamoe.mirai.contact.QQ;
import net.mamoe.mirai.event.events.bot.BotEvent;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * @author Him188moe
 */
public abstract class FriendEvent extends BotEvent {
    public final QQ sender;

    public FriendEvent(@NotNull Bot bot, @NotNull QQ sender) {
        super(bot);
        this.sender = Objects.requireNonNull(sender);
    }

    @NotNull
    public QQ getSender() {
        return sender;
    }
}

File: mirai-core/src/main/java/net/mamoe/mirai/event/events/qq/FriendEvent.kt
package net.mamoe.mirai.event.events.qq

import net.mamoe.mirai.Bot
import net.mamoe.mirai.contact.QQ
import net.mamoe.mirai.event.events.bot.BotEvent

/**
 * @author Him188moe
 */
abstract class FriendEvent(bot: Bot, val sender: QQ) : BotEvent(bot)